classes/Project/ProjectFactory.js

```diff
@@ -181,6 +184,284 @@ export default class ProjectFactory {
+        members: modules['Group Members'] ? members : { [creator]: { roles: [] } },
+        customRoles: modules['Group Members'] ? customRoles : {}
+      }
+    )
+    return copiedGroup
+  }
+
+  static async cloneLayers(project, copiedProject, database, withAnnotations = true) {
+    for (const layer of project.layers) {
+      const newLayer = {
+        id: `${process.env.SERVERURL}project/${copiedProject._id}/layer/${database.reserveId()}`,
+        label: layer.label,
+        pages: []
+      }
+
+      const newPages = await this.clonePages(layer, copiedProject, database, withAnnotations)
+      newLayer.pages.push(...newPages)
+      copiedProject.layers.push(newLayer)
+    }
+  }
+
+  static async clonePages(layer, copiedProject, database, withAnnotations) {
+    const newPages = await Promise.all(layer.pages.map(async (page) => {
+      if (withAnnotations) {
+        return await this.clonePagesWithAnnotations(layer, page, copiedProject, database)
+      }
+      return await this.clonePageWithoutAnnotations(page, copiedProject, database)
+    }))
+    return newPages
+  }
+
+  static async clonePageWithoutAnnotations(page, copiedProject, database) {
+    return {
+      id: `${process.env.SERVERURL}project/${copiedProject._id}/page/${database.reserveId()}`,
+      label: page.label,
+      target: page.target
+    }
+  }
+
+  static async clonePagesWithAnnotations(layer, page, copiedProject, database) {
+    if(!page.id.startsWith(process.env.RERUMIDPREFIX)) {
+      return {
+        id: `${process.env.SERVERURL}project/${copiedProject._id}/page/${database.reserveId()}`,
+        label: page.label,
+        target: page.target
+      }
+    }
+    else {
+      return await fetch(page.id)
+      .then(response => response.json())
+      .then(async pageData => {
+        const newPage = new Page(layer.id, {
+          id: `${process.env.SERVERURL}project/${copiedProject._id}/page/${database.reserveId()}`,
+          label: page.label,
+          target: page.target,
+          items: await Promise.all(pageData.items.map(async item => {
+            return await fetch(item.id)
+            .then(response => response.json())
+            .then(async itemData => {
+              const newItem = new Line({
+                id: `${process.env.SERVERURL}project/${copiedProject._id}/line/${database.reserveId()}`,
+                target: itemData.target,
+                body: itemData.body,
+                motivation: itemData.motivation,
+                label: itemData.label,
+                type: itemData.type
+              })
+              return await newItem.update()
+            })
+          }))
+        })
+        return await newPage.update()
+      })
+    }
+  }
+
+  static async copyProject(projectId, creator) {
+    if (!projectId) {
+      throw {
+        status: 400,
+        message: "No project ID provided"
+      }
+    }
+
+    const project = await ProjectFactory.loadAsUser(projectId, creator)
```

Comment:
Definitely don't `loadAsUser()` here. This should be a copy of the database object, not the projection for the User.